pkg/server/server.go

```diff
@@ -222,15 +246,161 @@ func (s *Server) Run(ctx context.Context) error {
+		ctx := req.Request.Context()
+		requestInfo, ok := apirequest.RequestInfoFrom(ctx)
+		if !ok {
+			responsewriters.ErrorNegotiated(
+				apierrors.NewInternalError(fmt.Errorf("no RequestInfo found in the context")),
+				// TODO is this the right Codecs?
+				scheme.Codecs, schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}, res.ResponseWriter, req.Request,
+			)
+			return
+		}
+
+		// If it's not the core group, pass through
+		if requestInfo.APIGroup != "" {
+			chain.ProcessFilter(req, res)
+			return
+		}
+
+		// Special handling of the core ("") group
+
+		if requestInfo.IsResourceRequest {
+			// This is a CRUD request for somethig like pods. Try to see if there is a CRD for the resource. If so, let the CRD
+			// server handle it.
+			crdName := requestInfo.Resource + ".core"
+			_, err := serverChain.CustomResourceDefinitions.Informers.Apiextensions().V1().CustomResourceDefinitions().Lister().GetWithContext(ctx, crdName)
+			if err == nil {
+				serverChain.CustomResourceDefinitions.GenericAPIServer.Handler.NonGoRestfulMux.ServeHTTP(res.ResponseWriter, req.Request)
+				return
+			}
+		} else if req.Request.URL.Path == "/api/v1" || req.Request.URL.Path == "/api/v1/" {
+			// This is a discovery request. We may need to combine discovery from the GenericControlPlane (which has built-in v1 types)
+			// and CRDs, if there are any v1 CRDs.
+
+			// If we are retrieving the GenericControlPlane's v1 APIResources, let that request through.
+			const passthroughHeader = "X-Kcp-Api-V1-Discovery-Passthrough"
+			if _, passthrough := req.Request.Header[passthroughHeader]; passthrough {
+				chain.ProcessFilter(req, res)
+				return
+			}
+
+			// Get all the CRDs (in the context's logical cluster) to see if any of them are in v1
+			crds, err := serverChain.CustomResourceDefinitions.Informers.Apiextensions().V1().CustomResourceDefinitions().Lister().ListWithContext(ctx, labels.Everything())
+			if err != nil {
+				// FIXME
+			}
+			crdDiscovery, ok := apiextensionsapiserver.APIResourcesForGroupVersion("", "v1", crds)
+			if !ok {
+				// No v1 CRDs - let the request through.
+				chain.ProcessFilter(req, res)
+				return
+			}
+
+			// v1 CRDs present - need to clone the request, add our passthrough header, and get /api/v1 discovery from
+			// the GenericControlPlane's server.
+			cr := utilnet.CloneRequest(req.Request)
+			cr.Header.Add(passthroughHeader, "1")
+
+			writer := newInMemoryResponseWriter()
+			serverChain.GenericControlPlane.GenericAPIServer.Handler.GoRestfulContainer.ServeHTTP(writer, cr)
+			if writer.respCode != http.StatusOK {
+				// FIXME
+			}
+
+			// Decode the response
+			into := &metav1.APIResourceList{}
+			obj, _, err := genericcontrolplane.Codecs.UniversalDeserializer().Decode(writer.data, nil, into)
+			if err != nil {
+				// FIXME
```

Comment:
Not sure how to handle this error & would love suggestions

Reply:
as above, fail hard

Reply:
Agreed